### PR TITLE
dev/core#5510 Fix retriving of the soft credit profile when generating…

### PR DESCRIPTION
**Pull request by seamuslee001**
… receipt

Overview
----------------------------------------
This fixes pulling out the soft credit profile id when generating receipt

Before
----------------------------------------
Potentially incorrect profile id is returned due to lacking of checking on the module + the lack of checking of the relevant entity_table

After
----------------------------------------
Correct profile for soft crediting is returned

@eileenmcnaughton @demeritcowboy

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
hmm  the goal was to move this stuff to the workflow template layer - I got it working on the participant side - https://github.com/civicrm/civicrm-core/pull/30374 but waiting on @JKingsnorth to be able to get back to testing that

**Comment by seamuslee001** [comment]
I'm not looking to move this code about, this just fixing a bug that our client had with the wrong/missing data in the receipt

**Comment by eileenmcnaughton** [comment]
@seamuslee001 well I'll put it on the long-list to take a look once the other is merged - I think this would be mergable as an interim fix if there were a test covering the relevant receipt contents

**Comment by seamuslee001** [comment]
@eileenmcnaughton I have worked on adding a unit test on this one but it seems like my test is failing in https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/BAO/ContributionPage.php#L314 I don't know why when i comment out those lines locally I can get an email to send its confusing me right now. It seems to be something to do with how that function is returning but not 100% sure why. It works in the UI tho

**Comment by seamuslee001** [comment]
@eileenmcnaughton now I have fixed the test set up issue and one other minor issue the UT passes for me locally, lets see what Jenkins thinks

File: tests/phpunit/CRM/Contribute/Form/Contribution/ConfirmTest.php [hunk 1]
```diff
@@ -922,4 +922,95 @@ public function testSubmitMultiIntervalMembershipContributionPage(bool $isQuickC
...
+      'id' => $this->getContributionPageID(),
+      'soft_credit_type_id' => 2,
+      'honor' => [
+        'first_name' => 'James',
+        'last_name' => 'Bond',
+        'street_address-1' => 'Vaxhaul Cross',
+        'city-1' => 'London',
+        'postal_code-1' => 'M4K1J1',
+        'country-1' => 1039,
+        'state_province-1' => 1068,
```
Comment (demeritcowboy) [inline tests/phpunit/CRM/Contribute/Form/Contribution/ConfirmTest.php:1002]: I assume the unit test fail is jenkins pointing out that this other James Bond that lives in London, Ontario, Canada appears to have a Toronto postal code! Crazy!

**Pull request merged**